external_project/models/project.py

```diff
@@ -0,0 +1,321 @@
+# -*- coding: utf-8 -*-
+# Copyright 2016 Akretion (http://www.akretion.com)
+# Benoit Guillot <benoit.guillot@akretion.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from odoo import fields, api, models
+from odoo.exceptions import Warning as UserError
+from odoo.tools.translate import _
+from odoo.tools.safe_eval import safe_eval
+from lxml import etree
+import requests
+import json
+
+
+ISSUE_DESCRIPTION = u"""Ce qui ne va pas:
+---------------------------
+
+
+Voilà comment cela devrait fonctionner:
+-----------------------------------------------------------
+"""
+
+
+class ExternalTask(models.Model):
+    _name = 'external.task'
+
+    @api.model
+    def _get_default_external_user(self):
+        return self.env.user
+
+    @api.model
+    def _get_origin(self):
+        if self.model_reference:
+            rec_name = self.model_reference._rec_name
+            if rec_name:
+                self.task_origin = self.model_reference.display_name
+
+    @api.model
+    def _authorised_models(self):
+        """ Inherit this method to add more models depending of your
+            modules dependencies
+        """
+        models = self.env['ir.model'].search(
+            [('model', '!=', 'external.task')])
+        return [(x.model, x.name) for x in models]
+
+    action_id = fields.Many2one(
```

Comment:
TODO this field are missing on internal_project. Moreover it can be hard to build the right path on our ERP. Maybe storing a url is a better solution?